website/www/site/content/en/documentation/ml/multi-language-inference.md

```diff
@@ -0,0 +1,155 @@
+
+### Compile Python code into package
+
+The Python code needs to be compiled into a package as a tarball. This package can then be used in the Java pipeline. The following example shows how to compile the Python code into a package:
+
+```bash
+ python setup.py sdist
+ ```
+
+In order to run this, a setup.py file must be made/
+
+### Run the Java pipeline
+The Java pipeline is defined in the `MultiLangRunInference` class. In this pipeline, the data is read from Google Cloud Storage, the cross-language Python transform is applied, and the output is written to Google Cloud Storage. The following example shows a snippet of the pipeline:
+
+```java
+Pipeline p = Pipeline.create(options);
+    PCollection<String> input = p.apply("Read Input", TextIO.read().from(options.getInputFile()));
+
+    List<String> local_packages=new ArrayList<String>();
+        local_packages.add("transformers==4.26.0");
+        local_packages.add("torch==1.13.1");
+        local_packages.add(options.getLocalPackage());
+
+    List<String> packages=new ArrayList<String>();
+        input.apply("Predict", PythonExternalTransform.<PCollection<String>, PCollection<String>>from(
+                "multi_language_custom_transform.composite_transform.InferenceTransform")
+                .withKwarg("model", options.getModelName())
+                .withKwarg("model_path", options.getModelPath())
+                .withExtraPackages(local_packages)
+                )
+                .apply("Write Output", TextIO.write().to(options.getOutputFile()));
+    p.run().waitUntilFinish();
+```
+
+The `PythonExternalTransform` is used to inject the cross-language Python transform into the Java pipeline. `PythonExternalTransform` takes a string parameter which is the fully qualified name of the Python transform.
+
+The `withKwarg` method is used to specify the parameters that are needed for the Python transform. In this example the `model` and `model_path` parameters are specified. These parameters are used in the initialization function of the composite Python PTransform, as shown in the first section. Finally the `withExtraPackages` method is used to specify the additional python packages that are needed for the Python transform. In this example the `local_packages` list is used, which contains the path to the compiled Python package.
+
+To run the pipeline, use the following command:
+
+```bash
+mvn compile exec:java -Dexec.mainClass=org.MultiLangRunInference \
+    -Dexec.args="--runner=DataflowRunner \
+                 --project=$GCP_PROJECT\
+                 --region=$GCP_REGION \
+                 --gcpTempLocation=gs://$GCP_BUCKET/temp/ \
+                 --inputFile=gs://$GCP_BUCKET/input/imdb_reviews.csv \
+                 --outputFile=gs://$GCP_BUCKET/output/ouput.txt \
+                 --modelPath=gs://$GCP_BUCKET/input/bert-model/bert-base-uncased.pth \
+                 --modelName=$MODEL_NAME \
+                 --localPackage=$LOCAL_PACKAGE" \
```

Comment:
We should clarify that we mean "Path to local package" here.

Reply:
Local is fine, but we should just clarify/document that this can be any package that can be resolved in a setup.py file.